LibreNMS/Util/Url.php

```diff
@@ -225,13 +225,7 @@ public static function sensorLink($sensor, $text = null, $type = null, $overlib
      */
     public static function deviceUrl($device, $vars = [])
     {
-        $routeParams = [is_int($device) ? $device : $device->device_id];
-        if (isset($vars['tab'])) {
-            $routeParams[] = $vars['tab'];
-            unset($vars['tab']);
-        }
-
-        return route('device', $routeParams) . self::urlParams($vars);
+        return self::generate(['page' => 'device', 'device' => is_int($device) ? $device : $device->device_id], $vars);
```

Comment:
This is going to cause issues, above code has a workaround for a bug with legacy code calling it.